src/main/java/com/butbetter/storage/custom_converter/BeanAddressConverter.java

```diff
@@ -0,0 +1,152 @@
+	 *
+	 *      an example for a correct String might look like this:
+	 *          ```Address{uuid=null, name='a', companyName='a', street='a', city='a', postCode='a', country='a'}```
+	 *
+	 *  2) matching and pulling of all the needed information, out if the given String where we now are quite sure,
+	 *  that it contains the information it needs
+	 *
+	 *      this process has to work with the constraints, the Java Regex Library put's on itself,
+	 *      since it's very easy to overwrite a match.
+	 *      Because of that, the method ```String getGroup(Matcher matcher, int i)```
+	 *      is needed to carefully get them out by checking first, so it's loaded in and can be extracted
+	 *      (for more information on that problem, read here:
+	 *      https://stackoverflow.com/questions/5674268/no-match-found-when-using-matchers-group-method#5674321)
+	 *
+	 *  3) takes the cleaned Strings from the now extracted information and puts them into the now ready Address Constructor
+	 *
+	 * @param value String of Address Object
+	 * @return Address parsed from given String
+	 *
+	 * @throws CsvDataTypeMismatchException thrown, if the given String wasn't in the right format,
+	 * and therefore cannot be converted to the needed Object
+	 *
+	 * @see AbstractBeanField
+	 */
+	@Override
+	public final Object convert(String value) throws CsvDataTypeMismatchException, CsvConstraintViolationException {
+		// Pattern check, if the String contains the needed value and is in the right format
+		if (!check_pattern.matcher(value).find()) {
+			CsvDataTypeMismatchException ex = new CsvDataTypeMismatchException();
+			logger.error(ex.getMessage());
+			throw ex;
+		}
+
+		// Matching of all needed information
+		Matcher name_matcher = name_pattern.matcher(value);
+		Matcher company_name_matcher = company_name_pattern.matcher(value);
+		Matcher street_matcher = street_pattern.matcher(value);
+		Matcher city_matcher = city_pattern.matcher(value);
+		Matcher postCode_matcher = postCode_pattern.matcher(value);
+		Matcher country_matcher = country_pattern.matcher(value);
+
+		String name = getCleanValue(
+				getGroup(name_matcher, 0),
+				NAME_IDENTIFIER + VALUE_PACKAGE,
+				VALUE_PACKAGE);
+
+		String company_name = getCleanValue(
+				getGroup(company_name_matcher, 0),
+				COMPANY_NAME_IDENTIFIER + VALUE_PACKAGE,
+				VALUE_PACKAGE);
+
+		String street = getCleanValue(
+				getGroup(street_matcher, 0),
+				STREET_IDENTIFIER + VALUE_PACKAGE,
+				VALUE_PACKAGE);
+
+		String city = getCleanValue(
+				getGroup(city_matcher, 0),
+				CITY_IDENTIFIER + VALUE_PACKAGE,
+				VALUE_PACKAGE);
+
+		String postCode = getCleanValue(
+				getGroup(postCode_matcher, 0),
+				POSTCODE_IDENTIFIER + VALUE_PACKAGE,
+				VALUE_PACKAGE);
+
+		String country = getCleanValue(
+				getGroup(country_matcher, 0),
+				COUNTRY_IDENTIFIER + VALUE_PACKAGE,
+				VALUE_PACKAGE);
+
+		return new Address(name, company_name, street, city, postCode, country);
```

Comment:
Do you want every time import and export generate a new UUID ? 

```java
 return new Address(name, company_name, street, city, postCode, country).setUUID(uuid);
```

Reply:
did change it to accommodate given [user story constrains](https://butbetter.atlassian.net/browse/STOR-40?atlOrigin=eyJpIjoiNTFmM2Y3MDc1YWZjNDgwNWFlOWQzN2EyNWU4ZjUyNGYiLCJwIjoiaiJ9)